sklearn/metrics/pairwise.py

```diff
@@ -185,7 +197,7 @@ def check_pairwise_arrays(
                 "(n_queries, n_indexed). Got (%d, %d) "
                 "for %d indexed." % (X.shape[0], X.shape[1], Y.shape[0])
             )
-    elif X.shape[1] != Y.shape[1]:
+    elif X.ndim == 2 and X.shape[1] != Y.shape[1]:
```

Comment:
Because we're not comparing `n_samples` but `n_features`. To be able to compute pairwise distances both arrays need to have the same number of features.

We could use `_num_features` but it errors on 1d arrays so I wanted to keep it simple and just enable the check when we enforce 2d arrays. I can replace the `if X.ndim == 2` by `if ensure_2d` if it makes the reason clearer but it's essentially the same.

Reply:
I think a comment would be enough there then.